x/auctions/keeper/bid.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v19/utils"
+	"github.com/evmos/evmos/v19/x/auctions/types"
+)
+
+// GetHighestBid gets the highest bid
+func (k *Keeper) GetHighestBid(ctx sdk.Context) *types.Bid {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.KeyPrefixBid)
+
+	if bz == nil {
+		return &types.Bid{
+			Sender: "",
+			Amount: sdk.NewCoin(utils.BaseDenom, sdk.ZeroInt()),
+		}
+	}
+
+	var bid types.Bid
+	k.cdc.MustUnmarshal(bz, &bid)
+	return &bid
+}
+
+// SetHighestBid sets the highest bid
+func (k *Keeper) SetHighestBid(ctx sdk.Context, sender string, amount sdk.Coin) {
+	store := ctx.KVStore(k.storeKey)
+	bid := &types.Bid{
+		Sender: sender,
+		Amount: amount,
+	}
+	bz := k.cdc.MustMarshal(bid)
+	store.Set(types.KeyPrefixBid, bz)
```

Comment:
@Vvaradinov the key should contain the address so that you can query the prev bids of the user

`prefix | address -> amount`

Reply:
see above, preferably we want to get this data from Events.

Reply:
no one uses events